core/src/main/java/feign/Request.java

```diff
@@ -82,6 +82,10 @@ public static Body empty() {
       return new Request.Body(null, null, null);
     }
 
+    public Charset charset() {
```

Comment:
Hrmmm, why?!

if `encoding` is null usually mean this is a binary protocol

Reply:
Thanks for flagging that.  I didn't consider that use case.  I'll make the changes.